source/module_hsolver/hsolver_pw.cpp

```diff
@@ -55,7 +46,11 @@ void HSolverPW<T, Device>::solve(hamilt::Hamilt<T, Device>* pHamilt,
     this->hamilt_ = pHamilt;
     // select the method of diagonalization
     this->method = method_in;
-    this->initDiagh(psi);
+    // report if the specified diagonalization method is not supported
+    if (this->method != "cg" && this->method != "dav" && this->method != "dav_subspace" && this->method != "bpcg")
+    {
+        ModuleBase::WARNING_QUIT("HSolverPW::solve", "This method of DiagH is not supported!");
+    }
```

Comment:
```suggestion
    const std::initializer_list<std::string> _m = {"cg", "dav", "dav_subspace", "bpcg"};
    if (std::find(std::begin(_m), std::end(_m), this->method) == std::end(_m))
    {
        ModuleBase::WARNING_QUIT("HSolverPW::solve", "This method of DiagH is not supported!");
    }
```
you can try this to avoid too long code line and make easy for future development

Reply:
thanks, I'll learn the skills.